src/snmp_agent.c

```diff
@@ -0,0 +1,1513 @@
+            DEBUG(PLUGIN_NAME ": Data set not found for '%s' type.", dd->type);
+            pthread_mutex_unlock(&g_agent->lock);
+            return SNMP_NOSUCHINSTANCE;
+          }
+
+          ret = uc_get_value_by_name(temp, &values, &values_num);
+
+          if (ret != 0) {
+            ERROR(PLUGIN_NAME ": Failed to get value for '%s'.", temp);
+            pthread_mutex_unlock(&g_agent->lock);
+            return SNMP_NOSUCHINSTANCE;
+          }
+
+          assert(ds->ds_num == values_num);
+          assert(i < values_num);
+
+          char data[DATA_MAX_NAME_LEN];
+          size_t data_len = sizeof(data);
+          ret = snmp_agent_set_vardata(data, &data_len, dd->oids[i].type,
+                                       dd->scale, dd->shift, &values[i],
+                                       sizeof(values[i]), ds->ds[i].type);
+
+          sfree(values);
+          values_num = 0;
+
+          if (ret != 0) {
+            ERROR(PLUGIN_NAME ": Failed to convert '%s' value to snmp data.",
+                  temp);
+            pthread_mutex_unlock(&g_agent->lock);
+            return SNMP_NOSUCHINSTANCE;
+          }
+
+          requests->requestvb->type = dd->oids[i].type;
+          snmp_set_var_typed_value(requests->requestvb,
+                                   requests->requestvb->type, data, data_len);
+
+          pthread_mutex_unlock(&g_agent->lock);
+
+          return SNMP_ERR_NOERROR;
+
+        } else {
+          DEBUG(PLUGIN_NAME ": %s:%d NOT IMPLEMENTED.", __FUNCTION__, __LINE__);
+        }
+      }
+    }
+  }
+
+  pthread_mutex_unlock(&g_agent->lock);
+
+  return SNMP_NOSUCHINSTANCE;
+}
+
+static int
+snmp_agent_table_index_oid_handler(struct netsnmp_mib_handler_s *handler,
+                              struct netsnmp_handler_registration_s *reginfo,
+                              struct netsnmp_agent_request_info_s *reqinfo,
+                              struct netsnmp_request_info_s *requests) {
+
+  if (reqinfo->mode != MODE_GET && reqinfo->mode != MODE_GETNEXT) {
+    DEBUG(PLUGIN_NAME ": Not supported request mode (%d).", reqinfo->mode);
+    return SNMP_ERR_NOERROR;
+  }
+
+  oid_t oid;
+
+  pthread_mutex_lock(&g_agent->lock);
+
+  memcpy(oid.oid, requests->requestvb->name,
+         sizeof(oid.oid[0]) * requests->requestvb->name_length);
+  oid.oid_len = requests->requestvb->name_length;
+
+  for (llentry_t *te = llist_head(g_agent->tables); te != NULL; te = te->next) {
+    table_definition_t *td = (table_definition_t *)te->value;
+
+    if (td->index_oid.oid_len &&
+          (snmp_oid_ncompare(oid.oid, oid.oid_len, td->index_oid.oid,
+                             td->index_oid.oid_len,
+                             MIN(oid.oid_len, td->index_oid.oid_len)) == 0)) {
+
+      DEBUG(PLUGIN_NAME ": Handle '%s' table index OID.", td->name);
+
+      int index = oid.oid[oid.oid_len - 1];
+      char *instance;
+
+      int ret = c_avl_get(td->index_instance, &index, (void **)&instance);
```

Comment:
Compound literals are great for when you need to pass a pointer but don't actually need about the value that's assigned. By using an anonymous variable, you're signaling to the reader that you don't care what's assigned.

```c
int ret = c_avl_get(td->index_instance, &index, &(void *){NULL});
```